cpp/include/raft/neighbors/detail/ivf_flat_build.cuh

```diff
@@ -42,6 +42,64 @@ namespace raft::neighbors::ivf_flat::detail {
 
 using namespace raft::spatial::knn::detail;  // NOLINT
 
+/**
+ * Update the state of the dependent index members.
+ */
+template <typename T, typename IdxT>
+void recompute_internal_state(const raft::resources& res, index<T, IdxT>& index)
```

Comment:
Is there anything different here compared to what we need to do for IVF-PQ? Can't we add this to `ivf_common.cuh`?